src/utils/setUserFromApiById.ts

```diff
@@ -0,0 +1,16 @@
+import { Dispatch } from 'react';
+import { getUser } from '../api';
+import { User } from '../types/User';
+
+export function setUserFromApiById(
+  userId: number,
+  setUser: React.Dispatch<React.SetStateAction<User | undefined>>,
+  setLoading: Dispatch<React.SetStateAction<boolean>>,
+) {
+  getUser(userId)
+    .then(setUser)
+    .catch(() => alert('User api is wrong!'))
```

Comment:
The error handling here uses an alert, which can be disruptive to the user experience. Consider using a more user-friendly error handling mechanism, such as displaying an error message within the UI.